openrank/apiserver/models.py

```diff
@@ -1,3 +1,126 @@
 from django.db import models
 
 # Create your models here.
+#All the Comments and doc-strings may contain Opinionated Views.
+#These are not the final docs. Constructive Negation is welcome!  ex  you may find the comments too-verbose
+
+class Contest(models.Model):
+    name=models.CharField()
+    description=models.TextField()
+    url = models.URLField()
+    is_public = models.BooleanField()
+    starttime = models.TimeField()
+    user_id = models.ForeignKey(User)
+    position = models.ForeignKey(Position)
+    cutoff_duration = models.IntegerField()
+    login_start_time = models.TimeField()
+    password = models.charField(max_length=45)
+
+
+class Position(models.models):
+	job_title = models.charField()
+
+
+
+
+
+class User(models.Model):
+	'''Currently these fiels are planned
+	1)name -eh just the name.     should there be a max length? for name? 
+	2)email - email of the contestant.  BuiltIn  EmailField from django is being used.
+	3)password-
+	4)phone_number- an IntegerField.    how will we validate this? is there any Buildin  (like how it is for Email)
+
+'''
+
+    role = models.OneToOneField()  # A Role can oprionally be assigned to a User. But a User MUST be assigned to SOME ROLE
+	name  = models.CharField(max_length=60)
+	email = models.EmailField(dmax_length=60)
+	password = models.CharField(max_length=50)
+	phone_number = models.charField(max_length=10)
+
+
+class Enrollment(models.Model)
+    user_id =    models.ForeignKey(User)
+    contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
+    marksobtained = models.IntegerField()
+    userDetails = models.charField()
+
+class Role(models.Model):
+	role = models.charField()
+
+
+class Question(models.Model):
+
+	description = models.TextField()
+	total_marks = models.IntegerField()
+	tags = models.manyToMany(Tag)
+
+	 
+
+
+class ContestQuestion(models.Model):
+	question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
+	contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
+	total_marks  = models.IntegerField()
+	time_limit   = models.IntegerField()
+	memory_limit = models.IntegerField()  
+	variable_constraints = models.CharField(max_length=45)
+
+
+class Language(models.Model):
+	language = models.charField()
+
+
+class QuestionLanguage(models.Model):
+	question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
+	language_id = models.ForeignKey(Language,on_delete=models.CASCADE)
+	function_signature =  models.TextField()
+	function_parameters = models.charField()
+	starter_code =        models.charField()
+
+
+class Tag(models.Model):
+	tag_description =   models.charField()  #A charField
+	questions = models.manyToMany(Question)
+    
+
+class TestCase(models.Model):
+	_input = models.charField()
```

Comment:
why the underscore? typo or intentional?

Reply:
input is  reserved
![image](https://user-images.githubusercontent.com/13259258/31109354-082ac806-a81d-11e7-839e-ac3cec18c9c7.png)

Reply:
ok. I wasn't aware. fine in that case. if you think testcase_input will be better use it otherwise _input is fine as well.